WordPress/Classes/Utility/ReaderPostCacheProvider.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+
+@objc class ReaderPostCacheProvider: NSObject {
+
+    private enum ReaderPostType {
+
+        case normal
+        case cross
+        case attribution
+
+        static let allTypes = [normal, cross, attribution]
+    }
+
+    let context: NSManagedObjectContext
+    init(context: NSManagedObjectContext) {
+        self.context = context
+    }
+
+    /**
+     Gets an stored ReaderPost from the local database, it searches for a Normal Post, 
+     a SourceAttribution Post or a Cross Post.
+     
+     - Parameters:
+        - postID: the postID of the post to fetch
+        - siteID: the siteID of the post to fetch, can be a blogID if it has source attribution information
+     
+     - Returns: A ReaderPost object if it is cached localy or nil if not found
+     */
+    func getPostByID(postID: Int, siteID: Int) -> ReaderPost? {
```

Comment:
I was thinking on that yesterday... I didn't want to polute the RemoteService method signature because for them it's transparent which kind of post it is, it just cares about the siteID and postID.

But I also though that I could use a block predicate and have the conditionals there something like:
```swift
let predicate = NSPredicate { (post, _) in
    return isNormalPost(post, postID:postID, siteID: siteID) || 
               isAttributionPost(post, postID:postID, siteID: siteID) || 
               isCrossPost(post, postID:postID, siteID: siteID)
}
```

That way I would only do one fetch request and won't polute the RemoteService signature.
What do you think?